pkgs/applications/blockchains/clightning/default.nix

```diff
@@ -1,23 +1,15 @@
-{ lib
-, stdenv
-, fetchurl
-, autoconf
-, automake
-, autogen
-, gettext
-, libtool
-, pkg-config
-, unzip
-, which
-, gmp
-, libsodium
-, python3
-, sqlite
-, zlib
-}:
+{ pkgs }:
 let
-  py3 = python3.withPackages (p: [ p.Mako p.mrkd ]);
-in
+  py3 = pkgs.python38.withPackages (p: with p; [ Mako mrkd ]);
+ 
+  # when building on darwin we somehow have to tell make where to find gsed
+  gsed = with pkgs; if stdenv.isDarwin then pkgs.gnused.overrideDerivation( attrs: {
```

Comment:
We should be patching usages of sed instead of overriding the derivation here. I see that you replaced the import list with `pkgs` so that you could do this, but I think this is pretty bad practice in nixpkgs.

Reply:
I suggest patching the package files to replace occurrences of `gsed` with `sed`.

Reply:
By patching the package files you mean using e.g. postPatch to override every occurrence of `gsed` with `sed` in all clightning files?

Reply:
done!